robottelo/hosts.py

```diff
@@ -1149,47 +1149,53 @@ def install(self, installer_obj=None, cmd_args=None, cmd_kwargs=None):
 
     def capsule_setup(self, sat_host=None, **installer_kwargs):
         """Prepare the host and run the capsule installer"""
-        self.satellite = sat_host or Satellite()
-        self.create_custom_repos(
-            capsule=settings.repos.capsule_repo,
-            rhscl=settings.repos.rhscl_repo,
-            ansible=settings.repos.ansible_repo,
-            maint=settings.repos.satmaintenance_repo,
-        )
-        self.create_custom_repos(rhel7=settings.repos.rhel7_os)
-        # self.execute('yum repolist')
+        self._satellite = sat_host or Satellite()
+
+        # Register capsule host to CDN and enable repos
+        result = self.register_to_cdn()
+        if result.status:
+            raise CapsuleHostError(f'Capsule CDN registration failed\n{result.stderr}')
```

Comment:
cmd.status check is already covered in  `register_to_cdn()`
```suggestion
        self.register_to_cdn()
```